### PR TITLE
repo: Add add-remotes-config-dir option

### PR DESCRIPTION
**Pull request by dbnicholson**
This option allows a repo to explicitly opt out of adding new remotes in
a remotes configuration directory. This currently defaults to true for
system repos and false for non-system repos to maintain legacy behavior
that non-system repos don't add remotes in a configuration directory.
That would be problematic for flatpak, which specifies a remotes config
dir but adds remotes in ways that are incompatible with it.

So, what this really does is allow system repos to control whether they
want to add remotes in the config dir or not. That's important if your
flatpak repo is the system repo like at Endless.

Closes: #1134

### CODE REVIEW

File: man/ostree.repo-config.xml [hunk 1]
```diff
@@ -122,6 +122,27 @@ Boston, MA 02111-1307, USA.
         keep free. The default value is 3.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>add-remotes-config-dir</varname></term>
```
Comment (cgwalters) [inline man/ostree.repo-config.xml]: Minor bikeshed...what do you think about `write-remotes-config-dir`?

Comment (dbnicholson) [reply man/ostree.repo-config.xml]: Oh, that would be fine. I meant to add in the PR that I was open to suggestions on the name.

**Comment by cgwalters** [comment]
No strong opinion here on the name, let's keep as is then.  Thanks for the patch!

@rh-atomic-bot r+ 4664c9f

**Comment by dbnicholson** [comment]
Heh, I was going to change it this morning. I just saw the email over the weekend but wasn't going to be working on it until today. I actually liked your name better.

**Pull request closed without merging**